internal/core/data/event.go

```diff
@@ -172,11 +172,10 @@ func deleteEventById(id string) error {
 
 // Delete the event and readings
 func deleteEvent(e contract.Event) error {
-	for _, reading := range e.Readings {
-		if err := deleteReadingById(reading.Id); err != nil {
-			return err
-		}
+	if err := dbClient.DeleteReadingsByDevice(e.Device); err != nil {
```

Comment:
This should not delete readings for the event's device. The function's purpose is to delete the event indicated by the incoming parameter. If there are ten events from the same device, and the request is to delete the first one, then this logic will wipe out readings for all ten of them. See utilized code paths here, none of them relate to the device.
<img width="461" alt="Screen Shot 2019-06-04 at 5 11 48 PM" src="https://user-images.githubusercontent.com/1342980/58914264-112bd580-86ec-11e9-91c0-18c42575646c.png">